src/Google.Cloud.Functions.Invoker/FunctionEnvironment.cs

```diff
@@ -126,12 +126,25 @@ internal void ConfigureServices(IServiceCollection services)
 
         /// <summary>
         /// Executes a request by asking the dependency injection context for an IHttpFunction, then
-        /// executing it. Any additional "root" handling (e.g. extra logging, or exception handling).
+        /// executing it. Any additional "root" handling (e.g. extra logging, or exception handling) goes here.
         /// </summary>
         internal async Task Execute(HttpContext context)
         {
-            var function = context.RequestServices.GetRequiredService<IHttpFunction>();
-            await function.HandleAsync(context);
+            try
+            {
+                var function = context.RequestServices.GetRequiredService<IHttpFunction>();
+                await function.HandleAsync(context);
+            }
+            catch (FunctionException ex)
```

Comment:
Maybe catch everything else as well to log a rethrow?

Reply:
No need for that - any exceptions that make it to the very top of the pipeline are logged and the response becomes a 500.